### PR TITLE
CH-464: add support for GH actions release builds

### PR DESCRIPTION
**Pull request by DanielMonteiroFRVR**
# Type of PR

[ ] Bug Fix
[X] New Feature
[ ] Other

# References

CH-464

# What problem does this PR address

We want to automate-away the (rather error-prone) release build process, having a Single Source of Truth™

# How does this PR addresses the problem

By having the build happening on a clean install of the build environment and centralised into a Github action, we go one step further, just requiring the upload of the resulting AAB into Google Play.

# Implementation notes

For now, this assumes always the develop versions of both tools and internal. Hopefully, we can expand from this very soon, allowing to specify the custom versions of both.

This solution was originally derived from the work done for iOS, which has much more complex requirements. Some extra steps are usually required for it, that are not required for Android platforms. Hopefully, you can point me to aspects not needed here!

# Platforms/Channels

This addresses Google Play only, for now.

# What was tested

Builds of Darts resulted in valid AAB files.
https://github.com/frvraps/game-darts/actions/runs/2005273646

# How to validate current changes

Generate Darts builds from its Action panel.

### CODE REVIEW

File: .github/workflows/build-google_play.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      deploy_msg: ${{ steps.metadata.outputs.deploy_msg }}  
+
+  ## --- Generate Google Play project for game ---
+  generate_googleplay:
+    name: Generate Google Play project
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    needs: [gather_game_data]
+    container:
+      image: ghcr.io/frvraps/fst@sha256:460d5348a9294ab18e907fa8480971bc765036981ca8337079988e36eaabde99
```
Comment (DanielMonteiroFRVR) [inline .github/workflows/build-google_play.yml]: This image is a snapshot of the tools and internal and might be updated before the merge

File: .github/workflows/build-google_play.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      deploy_msg: ${{ steps.metadata.outputs.deploy_msg }}  
+
+  ## --- Generate Google Play project for game ---
+  generate_googleplay:
+    name: Generate Google Play project
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    needs: [gather_game_data]
+    container:
+      image: ghcr.io/frvraps/fst:2.3.111
```
Comment (dnunesfrvr) [inline .github/workflows/build-google_play.yml]: We should all get the lasted version of the container or at least have the option to configure it

**Comment by RuiGamitoFRVR** [comment]
@DanielMonteiroFRVR and @dnunesfrvr

Is this PR in a ready to merge state? I always assumed it wasn't because the frvr-gh-workflows was not PRed, nor the entry point (PIPELINE_qa_builds_entrypoint.yml) on game-darts! :(
Anyway, seems there should be a major cleanup on both this and frvr-gh-workflows.

If the work here is done (as well as on the other repos) I'll start looking at this tomorrow.

**Comment by DanielMonteiroFRVR** [comment]
Right now, we're getting errors with googleapis, used for publishing entries to the masterdoc, but the build works fine. However, this failure with the masterdoc uploader is enough to make the job fail.

Also, since Android builds now require adaptive icons, I'm not using darts as a testbed anymore, but rather Drag Race (which has adaptive icons ready, along with Gold Digger and Basketball). I'll be publishing a PR very soon, along with a PR for the frvr-gh-workflows repo.

**Pull request closed without merging**